vpr/src/server/zlibutils.cpp

```diff
@@ -0,0 +1,83 @@
+#ifndef NO_SERVER
+
+#include "zlibutils.h"
+
+#include <cstring> // Include cstring for memset
+#include <zlib.h>
+
+std::optional<std::string> try_compress(const std::string& decompressed) {
+    z_stream zs;
+    memset(&zs, 0, sizeof(zs));
+
+    if (deflateInit(&zs, Z_BEST_COMPRESSION) != Z_OK) {
+        return std::nullopt;
+    }
+
+    zs.next_in = reinterpret_cast<Bytef*>(const_cast<char*>(decompressed.data()));
+    zs.avail_in = decompressed.size();
+
+    int ret_code;
+    char* result_buffer = new char[32768];
+    std::string result;
+
+    do {
+        zs.next_out = reinterpret_cast<Bytef*>(result_buffer);
+        zs.avail_out = sizeof(result_buffer);
+
+        ret_code = deflate(&zs, Z_FINISH);
+
+        if (result.size() < zs.total_out) {
+            result.append(result_buffer, zs.total_out - result.size());
+        }
+    } while (ret_code == Z_OK);
+
+    delete[] result_buffer;
+
+    deflateEnd(&zs);
+
+    if (ret_code != Z_STREAM_END) {
+        return std::nullopt;
+    }
+
+    return result;
+}
+
+std::optional<std::string> try_decompress(const std::string& compressed) {
+    z_stream zs;
+    memset(&zs, 0, sizeof(zs));
+
+    if (inflateInit(&zs) != Z_OK) {
+        return std::nullopt;
+    }
+
+    zs.next_in = reinterpret_cast<Bytef*>(const_cast<char*>(compressed.data()));
+    zs.avail_in = compressed.size();
+
+    int ret_code;
+    char* result_buffer = new char[32768];
```

Comment:
Same as above.

Reply:
fixed https://github.com/verilog-to-routing/vtr-verilog-to-routing/pull/2526/commits/f35de4ee90058597f73ddbc3d40d68abe9593d70